### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by diogorborges**

### CODE REVIEW

**Comment by arctouch-thiagocechetto** [comment]
Hi @diogorborges , thank you for submitting your app!

Why are the tests commented?

File: app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                return false;
+            }
+        });
+    }
+
+    @Override
+    public void onBackPressed() {
+        presenter.onViewStarted(this);
```
Comment (arctouch-thiagocechetto) [inline app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MainActivity.java]: Is there any special reason change `onBackPressed`beahvior?

File: app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MoviesAdapter.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                if (genre.id.equals(id)) {
+                    genresString.add(genre.name);
+                }
+            }
+        }
+        return StringUtils.join(genresString, ", ");
+    }
+
+    @NonNull
+    private String getFullImageUrl(Movie movie) {
```
Comment (arctouch-andrecardoso) [inline app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MoviesAdapter.java]: This method is duplicated on `DetailActivity`. How would you fix that?

File: app/src/test/java/com/diogorborges/themoviedbsample/presentation/detail/DetailPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+//import org.mockito.MockitoAnnotations;
+//
+//import static org.mockito.Matchers.anyInt;
+//import static org.mockito.Mockito.verify;
+//import static org.mockito.Mockito.when;
+//
+///**
+// * Created by polbins on 25/02/2017.
+// */
+//public class DetailPresenterTest {
```
Comment (arctouch-andrecardoso) [inline app/src/test/java/com/diogorborges/themoviedbsample/presentation/detail/DetailPresenterTest.java]: Why are the tests commented out?

File: app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MainActivity.java (hunk 1 above)
Comment (diogorborges) [reply app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MainActivity.java]: I did the tests before implement RX, instead I used callbacks as Use cases, I need to improve this, most likely on wednesday after a college test

About the backpressed, was not a smart code, this behavior was made to back to the main list of movies after you search for a movie

File: app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MoviesAdapter.java (hunk 2 above)
Comment (diogorborges) [reply app/src/main/java/com/diogorborges/themoviedbsample/presentation/main/MoviesAdapter.java]: Yes, actually I intent to change all of this image logic part (like glide image download) to the presenter

**Comment by arctouch-andrecardoso** [comment]
Hi @diogorborges ,

Thanks for submitting your code! We're done with reviewing.

Good luck! 🍀

**Pull request merged**